### ISSUE TITLE
Please help me with LOD Initial Cell Width. Too many polygons

### ISSUE THREAD

**Issue by MihaSmit** [spimort/TerraBrush#87]
I want the initial grid to be larger than the default.
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/dcf8e3b0-2caf-4685-866e-b3d98b1bf452)

I use the LOD Initial Cell Width setting to increase the width of the initial mesh. But the number of primitives does not change.
And when the camera moves, these edges constantly “float”
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/49db3a6d-e355-4f00-9407-08f69cd060f7)

![изображение](https://github.com/spimort/TerraBrush/assets/16068635/28406228-0f53-467d-8662-6d847a622779)

**Comment by spimort** [comment]
I'm not quite sure what you mean by the edges float but for the number of primitives, if you increase the initial cell width, you sould probably decrease the "levels" and/or the rows per level. Right now, in your example, it still creates 5 levels of details that are not even visible because you have too many rows per level and each levels are too large to be displayed. The primitives exists but are not visible.

For example, if I reduce the levels to 3 here, the number of primitives goes down and a maximum of 3 levels will be visible : 
![image](https://github.com/spimort/TerraBrush/assets/6035485/25c55741-9349-41ef-bd5d-fb23720a50fb)

If I change the value of levels to 5, the primitives goes up but are not being display because they are out of bounds : 
![image](https://github.com/spimort/TerraBrush/assets/6035485/0ea07c0c-65c2-4241-828f-d90d25f60e78)

So, you should change these numbers depending on the kind of terrain you want 🙂

I hope this helps!

**Comment by MihaSmit** [comment]
Sorry, you probably misunderstood me. I'll ask the question along the way. When I increase the LOD cell initial width values, the number of primitives is not removed. It's as if the visual mesh is getting wider, but in fact the polygons then remember what they were, if you set the original LOD cell width to 1, they will return to their position.
First, when creating the landscape, I needed to increase additional polygons so that there would be no transfers when carrying the camera. How can this be done?
P.S Google translator

**Comment by MihaSmit** [comment]
I want to make the world in a low poly style. I need the primitives to be bigger, not visually, but "really", or I don't understand something, but I tried everything and everything brings me back to the fact that I need to increase the width of the primitives without changing the values of LOD Initial Cell Width

**Comment by MihaSmit** [comment]
Yes, thanks, I figured out the primitives with your help, among other things. But now I don't understand why when I have a LOD Initial Cell Width above 1, the landscape is distorted even if it is in the right square? + if I specify the brush size 1 and make a rock, then it will not be visually visible, but if I return the LOD Initial Cell Width to 1, then I will see this rock. Can there be problems with this later with collision? It seems to me that something is wrong here. I don't understand, I just want to make the initial grid of the world wider :(

**Comment by spimort** [comment]
Ohh I see what you mean, with the video. Let me check if I can do something about it!

**Comment by spimort** [comment]
I've done a fix, can you tell me if the version 0.5.1Alpha fixes your issue 🙂?

**Comment by MihaSmit** [comment]
Thank you so much! I also noticed that if you set the brush size to 1 and draw, then the grid does not change. LOD Initial Cell Width is at 4 at this moment. And if you return LOG Initial Cell Width to 1, then these small mountains will appear. As I understand it, it's stored somewhere in memory, I'm not sure if it's good. After all, I will no longer set the LOD Initial Cell Width to 1, but the system remembers that there should be a small mountain there. I don't know how this will affect performance in the future. Could there be problems with a collision in the future because of this?

**Comment by MihaSmit** [comment]
And now, when I add new areas, the error All images must share the same format has appeared, although I do not have any textures. And also, when a new section is spawned, it inherit the geometry of the original region and I can no longer change the grid
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/86aa3a76-bb55-4f1c-a195-81af88a04551)

**Comment by MihaSmit** [comment]
And I also noticed that you fixed a bug with the fact that the edges "floated" in the first region, but in the subsequent ones it remained, in the second region they also float, it turns out that at a distance from the player all the mountains will "jump" with each update of the grid, even if they do not change the region

**Comment by spimort** [comment]
> Thank you so much! I also noticed that if you set the brush size to 1 and draw, then the grid does not change. LOD Initial Cell Width is at 4 at this moment. And if you return LOG Initial Cell Width to 1, then these small mountains will appear. As I understand it, it's stored somewhere in memory, I'm not sure if it's good. After all, I will no longer set the LOD Initial Cell Width to 1, but the system remembers that there should be a small mountain there. I don't know how this will affect performance in the future. Could there be problems with a collision in the future because of this?

The way it works is that you still draw pixels on an image that is the size of the terrain zone. If you increase the initial cell width, it skips pixels on the image but the pixels are still there. That's why you still see what you've done when going back to 1.
The prefered initialcellwidth will always be 1 because it's the easiest one to compute, since 1 pixel on the image equals to 1 meter of the mesh. Keep in mind that even if you change the initialcellwidth, the collision will still be based on 1 because I don't have much control on that on the heightmap collision shape.

The option is there to accomodate but the 1 will still be the best option.

I was wondering, is there a reason why you don't want to use the default 1? Visually, you can get the same result and in term of perf, 1 is really not that bad in my opinion.

**Comment by spimort** [comment]
> And now, when I add new areas, the error All images must share the same format has appeared, although I do not have any textures. And also, when a new section is spawned, it inherit the geometry of the original region and I can no longer change the grid ![изображение](https://private-user-images.githubusercontent.com/16068635/328024311-86aa3a76-bb55-4f1c-a195-81af88a04551.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTQ5MzQyOTAsIm5iZiI6MTcxNDkzMzk5MCwicGF0aCI6Ii8xNjA2ODYzNS8zMjgwMjQzMTEtODZhYTNhNzYtYmI1NS00ZjFjLWExOTUtODFhZjg4YTA0NTUxLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA1VDE4MzMxMFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU0YjYwZTA1NTA0ZTdlYTljYjNkMzU0ZjMzZDNkZTY5ZmEyNDY4YzZmYTNjZjBmYTRiMTcxYmJhMGU0ZWNmNWUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.Ie6BM5Cf59TGn5acZdN6vlmwT_E1t8zISSc3Z290ENo)

That is strange, is there any way you can share the project so I can test it?

**Comment by MihaSmit** [comment]
I don't want to use the number 1, since my game assumes low poly graphics, and also works on android, I try to save polygons. and I don't want to see a smooth and detailed surface.
I will reassemble the project a little later to make sure that I did everything correctly and write off whether the error will reappear.

**Comment by MihaSmit** [comment]
The problem with "All images must share the same format" went away after installing net 8 and recreating the project. Here is a video about floating faces in the second and next zone
https://www.youtube.com/watch?v=sgd73C_D9tc

**Comment by spimort** [comment]
I think you are right, there is an issue here!
I think I know how to solve it, I should do a fix soon 🙂

**Comment by spimort** [comment]
The version 0.5.3Alpha should fix it, please let me know if it works for you! 🙂

**Comment by MihaSmit** [comment]
Thank you very much! You've been very helpful! You are very good! Let me ask you a couple more things that I would suggest. At the moment, the collision does not match the grid as expected, you talked about it. Is it possible to come up with a way to improve the collision system if the standard specifies a value higher than 1 for "LOAD Initial Cell Width"? And I also noticed what's in the screenshot, I saw your video, I'll try to review it to figure out how to fix it. Everything else is fire! I really like it, everything is simple and convenient!
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/cde2abff-75e9-4f3c-b6a7-3decf37cbc8d)

**Comment by MihaSmit** [comment]
Yes, I double-checked these lines and they were not commented out in the code, as in your video, maybe me need to correct something here?
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/55db2ff4-0553-4f35-a755-f04dfccdf889)

**Comment by spimort** [comment]
> Thank you very much! You've been very helpful! You are very good! Let me ask you a couple more things that I would suggest. At the moment, the collision does not match the grid as expected, you talked about it. Is it possible to come up with a way to improve the collision system if the standard specifies a value higher than 1 for "LOAD Initial Cell Width"? And I also noticed what's in the screenshot, I saw your video, I'll try to review it to figure out how to fix it. Everything else is fire! I really like it, everything is simple and convenient! ![изображение](https://private-user-images.githubusercontent.com/16068635/328403102-cde2abff-75e9-4f3c-b6a7-3decf37cbc8d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTUxMzczNDgsIm5iZiI6MTcxNTEzNzA0OCwicGF0aCI6Ii8xNjA2ODYzNS8zMjg0MDMxMDItY2RlMmFiZmYtNzVlOS00ZjNjLWI2YTctM2RlY2YzN2NiYzhkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA1MDglMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwNTA4VDAyNTcyOFomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTdiODQ3OGFmNmFlMDRlZjRmZDM0ODgyZWIxNWZlMDBlMjBhNDJkNDg3OTUzMWJjOTIxNGM4MDM2NzAyNTcwNzEmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.1i5Fh5GAwz58Ta2gkTxJHoy9u2J0_Y75Rqp1sGucup0)

Sadly, changing the size of the cells in the collision shape is not possible (as far as I know) 😔. I use the heightmapshape3D from godot and I don't think there's much I can do about it 🫤

https://docs.godotengine.org/en/stable/classes/class_heightmapshape3d.html

I'll try to take a look the the weird line, it feels like it's the average zone transition calculation that does that. We'll see.

**Comment by MihaSmit** [comment]
please see if it is possible to multiply the width of the collision or apply any mathematical actions to it so that the collision at least approximates the meshes of the landscape, if possible. Thank you very much for your addon, we really like it and we will use it in our first projects!
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/5a002269-25f8-44e9-bc8a-45d4a89fa638)
![изображение](https://github.com/spimort/TerraBrush/assets/16068635/4ec036d9-6578-487e-9b7d-6341772a3d78)

**Comment by spimort** [comment]
hey @MihaSmit, if you want to try it before it goes live (and give me feedbacks), this should be possible now with this branch : 
https://github.com/spimort/TerraBrush/tree/feature/terrain-resolution (PR : https://github.com/spimort/TerraBrush/pull/173)

Feel free to ask me questions about it if needed.

**Comment by spimort** [comment]
This is now available with the version 0.11.0Alpha, I'll close the issue 🙂
